### DISCUSSION TITLE
Keycloak 22.0.5 - unable to uninstall a custom provider (jar)

### DISCUSSION THREAD

**Discussion by karnno** [keycloak/keycloak#25856] [Archive]
Hello, 

I write you today because I am blocked, even though I tried to followed the documentation. 

So, I have Keycloak 22.0.5 installed on Windows 11, and I start it with the "kc.bat start-dev" command.

Recently I installed a new provider via a JAR file that I copied in ./providers folder, then I typed the "kc.bat build" command

this documentation:
[https://www.keycloak.org/server/configuration-provider](https://www.keycloak.org/server/configuration-provider)

tells me that I can disable or uninstall my provider.

Uninstalling is done by removing the jar and running the build command again.

Unfortunately, when I do that I have an exception:
Caused by: java.nio.file.NoSuchFileException: C:\Users\noviseka\javatools\keycloak-22.0.5\lib\..\providers\my-remote-mysql-user-storage-provider.jar

Also, I tried to disable it  using the command in the documentation, but no success, when I go into my Keycloak admin interface, in the "user federation" page, I can still add a provider using my custom implementation.

Can you please help ? 

Best regards,
Karnno

**Comment by Captain-P-Goldfish** [comment]
For all who get the same problem in the future. This is a problem with the custom-artifact build. It happens if the custom-provider has dependencies packaged that are already present within keycloak itself. Set these dependencies to provided and keep them out of your artifact and you should be able to uninstall the custom-provider again.

**Comment by mabartos** [comment]
@karnno Could you please try it with the more recent Keycloak release? And it'd be great to share the whole stack trace.

**Reply by mabartos** [reply]
Answered by @Captain-P-Goldfish

**Comment by mabartos** [comment]
Answered by @Captain-P-Goldfish
